src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,108 @@
+    IStrategyManager public immutable strategyManager;
+
+    StrategyBase public immutable strategyImplementation;
+
+    ProxyAdmin public eigenLayerProxyAdmin;
+
+    // @notice Mapping token => strategy contract for the token
+    mapping(IERC20 => IStrategy) public tokenStrategies;
+
+    event ProxyAdminChanged(ProxyAdmin previousProxyAdmin, ProxyAdmin newProxyAdmin);
+    event StrategySetForToken(IERC20 token, IStrategy strategy);
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager, StrategyBase _strategyImplementation) {
+        strategyManager = _strategyManager;
+        strategyImplementation = _strategyImplementation;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        ProxyAdmin _eigenLayerProxyAdmin
+    )
+        public virtual initializer
+    {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        // TODO: decide if a function for changing this is warranted
+        eigenLayerProxyAdmin = _eigenLayerProxyAdmin;
+        emit ProxyAdminChanged(ProxyAdmin(address(0)), _eigenLayerProxyAdmin);
+    }
+
+    // TODO: document with ample warnings
+    function deployNewStrategy(IERC20 token) external onlyWhenNotPaused(PAUSED_NEW_STRATEGIES) {
+        require(tokenStrategies[token] == IStrategy(address(0)),
+            "StrategyFactory.deployNewStrategy: Strategy already exists for token");
+        IStrategy strategy = IStrategy(
+            address(
+                new TransparentUpgradeableProxy(
+                    address(strategyImplementation),
+                    address(eigenLayerProxyAdmin),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, token, pauserRegistry)
+                )
+            )
+        );
+        _setStrategyForToken(token, strategy);
+    }
+
+    /** 
+     * @notice Owner-only function to pass through a call to `StrategyManager.addStrategiesToDepositWhitelist`
+     * @dev Also adds the `strategiesToWhitelist` to the `tokenStrategies` mapping
+     */
+    function whitelistStrategies(
+        IStrategy[] calldata strategiesToWhitelist,
+        bool[] calldata thirdPartyTransfersForbiddenValues
+    ) external onlyOwner {
+        strategyManager.addStrategiesToDepositWhitelist(strategiesToWhitelist, thirdPartyTransfersForbiddenValues);
+        for (uint256 i = 0; i < strategiesToWhitelist.length; ++i) {
+            IERC20 underlyingToken = strategiesToWhitelist[i].underlyingToken();
+            _setStrategyForToken(underlyingToken, strategiesToWhitelist[i]);
+        }
+    }
+
+    // @notice Owner-only function to add (existing) Strategy contracts to the `tokenStrategies` mapping
+    function editTokenStrategiesMapping(
```

Comment:
tbh just put in initializer?

Reply:
it could also be in the initializer, but I think this is desirable regardless

Reply:
function removed in https://github.com/Layr-Labs/eigenlayer-contracts/pull/522/commits/b88cdb5617d8ba51b15b6472f655d8043d1929ee so this discussion is no longer relevant